### DISCUSSION TITLE
mpas elevation check

### DISCUSSION THREAD

**Discussion by hkershaw-brown** [NCAR/DART#332] [General]
What needs to be checked in mpas for surface elevations?  
When should observations be rejected?

Here is the comment in https://github.com/NCAR/DART/pull/95

![Screen Shot 2020-12-14 at 9 38 15 AM](https://user-images.githubusercontent.com/20047007/102108449-2e1c8b00-3df0-11eb-9d66-dba51459f140.png)

**Comment by nancycollins** [comment]
i think QTY_SURFACE_ELEVATION should bypass the max diff tests regardless of the setting.

i don't know about QTY_SURFACE_PRESSURE - i would think that should be treated like any other obs and rejected if the diff is too large, but maybe that is a soyoung question.

just for the record for everything EXCEPT surface elevation:

both the wrf and mpas model_mods already have (and have had for a long time) a namelist variable which should be set to -1 to disable the distance check for all QTYs, and set to a positive value to set the max difference in meters that is ok.  this is used when the interpolation location vert type is VERTISSURFACE.  the vertical value in the location derived type is expected to be the elevation of the observation at that location, in meters.   the model_mod computes the model's surface elevation at the obs location and compares the values. a larger difference than the threshold is rejected as a failed forward operator/interpolation.

the namelist item is:  input.nml :: &model_nml :: sfc_elev_max_diff 

which is an integer value, in meters.  -1 disables the check.

**Comment by timhoar** [comment]
I still feel the need to discuss this. The first `if` block **skips** the tolerance test for all kinds of "surface" quantities.
If we remove most of them, then they will fall into the `else` block and get tested. The RTTOV FO code sets
the vertical values such that the test in the `else` block is moot. The RTTOV interpolations will always pass
the test ... good.

However, all of the other forward operators do not have this guarantee, and the `COMMON_CODE` forward
operators (e.g. `!RADIOSONDE_SURFACE_PRESSURE,  QTY_SURFACE_PRESSURE,     COMMON_CODE`)
have the vertical value as specified by the observation conversion.

```
if((obs_kind == QTY_SURFACE_PRESSURE) .or. (obs_kind == QTY_SURFACE_ELEVATION)    .or. &
   (obs_kind == QTY_2M_TEMPERATURE)   .or. (obs_kind == QTY_2M_SPECIFIC_HUMIDITY) .or. &
   (obs_kind == QTY_SKIN_TEMPERATURE) .or. (obs_kind == QTY_SURFACE_TYPE)         .or. &
   (obs_kind == QTY_CLOUD_FRACTION)   .or. &
   (obs_kind == QTY_10M_U_WIND_COMPONENT) .or. (obs_kind == QTY_10M_V_WIND_COMPONENT)) then
    istatus = 0
   if ( debug > 1 .and. do_output()) print *, &
   'model_interpolate: pass sfc_elev_max_diff check for ',trim(get_name_for_quantity(obs_kind))
else
   ! Reject obs if the station height is far way from the model terrain.
   if(is_vertical(location, "SURFACE") .and. sfc_elev_max_diff >= 0) then
   if(abs(llv(3) - zGridFace(1,cellid)) > sfc_elev_max_diff) then
      istatus = 12
      if (debug > 0 .and. do_output()) then
         print*, 'model_interpolate: Skip due to abs(dz) > sfc_elev_max_diff:', &
         llv(3)-zGridFace(1,cellid), trim(get_name_for_quantity(obs_kind)),' at ',trim(locstring)
         goto 100
      endif
   endif
   endif
endif
```

The block of code used to (circa Aug 11, prior to RTTOV) look like (0c50fcbffa7096bf57da8ebeb54053d408b0ecd7):

```
! Reject obs if the station height is far way from the model terrain.
if(is_vertical(location, "SURFACE").and. sfc_elev_max_diff >= 0) then
   if(abs(llv(3) - zGridFace(1,cellid)) > sfc_elev_max_diff) then
      ! allow experimentation with always accepting surface altimeters
      ! by setting this module global variable to .true. at top of file.
      if (always_assim_surf_altimeters .and. &
          (obs_kind == QTY_SURFACE_PRESSURE .or. obs_kind == QTY_SURFACE_ELEVATION)) then
         istatus = 0
      else
         istatus = 12
         if (debug > 10 .and. do_output()) &
         print*, 'model_interpolate: Skip due to abs(dz) > sfc_elev_max_diff:', llv(3)-zGridFace(1,cellid)
         goto 100
      endif
   endif
endif
```
It checked surface pressure and surface elevation. It also uses a module variable `always_assim_surf_alitimeters`, which is still in the code - and defaults to .true. 

Are we simply saying that we should revert to the 0c50fcbffa7 block of code?

**Comment by nancycollins** [comment]
yes.  :)

p.s. when we were looking at the code yesterday, i noticed that it sets a logical "surface_obs" a few lines up.  so the first line of that block could be a bit more readable:

if(surface_obs .and. sfc_elev_max_diff >= 0) then ...

**Comment by nancycollins** [comment]
actually - let me amend that last answer.  the block should mostly be the same, but i think it needs to skip the test for surface elevation in *all cases*, not just when always_assim_surf_altimeters is true.  this would be a slight change from the previous code, but it seems to me if the caller is asking for the surface elevation they don't know it already so they can't pass it in with the location.

edit to add:  i mean this:

if (always_assim_surf_altimeters .and. &
          (obs_kind == QTY_SURFACE_PRESSURE .or. obs_kind == QTY_SURFACE_ELEVATION)) then

changes to:

if ((always_assim_surf_altimeters .and. obs_kind == QTY_SURFACE_PRESSURE) .or . &
      obs_kind == QTY_SURFACE_ELEVATION) then

**Comment by hkershaw-brown** [comment]
moving this to discussions since no decision on this.
